dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1362,118 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _update_correlation(self, clean_samples):
+        """
+        Update correlation matrix for cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if self._profile[col].profile['data_type'] not in ['int', 'float'] \
+                or self._profile[col].profile['statistics']['null_count'] > 0:
```

Comment:
this needs to be tabbed over twice, i also suggest if you put `()` around the entire statement  you won't need the `\`

Reply:
Also, I'm concerned that this is using profile. Maybe it would be better to open up the API for the compiler to allow it to return the `data_type` based on current state as a functional property

Reply:
potentially a future PR revision though

Reply:
agree, and some statistics as well

Reply:
does the null count need to be acquired from the profile or can we get it directly from the class property?

Reply:
any suggestion of how to get it from the class?

Reply:
looks like they have a property called `null_count`

Reply:
changed to use with class property